code/modules/hivemind/machines.dm

```diff
@@ -0,0 +1,486 @@
+
+
+/obj/structure/hivemind_machine/emp_act(severity)
+	switch(severity)
+		if(1)
+			get_damaged(30)
+			stun(10)
+		if(2)
+			get_damaged(10)
+			stun(5)
+
+
+/////////////////////////////////////////////////////////////
+/////////////////////////>MACHINES<//////////////////////////
+/////////////////////////////////////////////////////////////
+
+//CORE-GENERATOR
+//generate evopoints, spread weeds
+/obj/structure/hivemind_machine/node
+	name = "strange hive"
+	desc = "That's definitely not a big brother, but it's still watching you."
+	max_health = 320
+	icon_state = "core_new"
+	//internals
+	var/list/my_wireweeds = list()
+
+/obj/structure/hivemind_machine/node/Initialize()
+	if(!hive_mind_ai)
+		hive_mind_ai = new /datum/hivemind
+	..()
+
+	hive_mind_ai.hives.Add(src)
+
+	update_icon()
+
+	var/obj/effect/plant/hivemind/founded_wire = locate() in loc
+	if(!founded_wire)
+		var/obj/effect/plant/hivemind/wire = new(loc, new /datum/seed/wires)
+		add_wireweed(wire)
+		wire.Process()
+	else
+		for(var/obj/effect/plant/hivemind/W in range(6, src))
+			if(W.master_node)
+				if(!(locate(type) in W.loc) && (get_dist(W, W.master_node) > 6) )
+					add_wireweed(W)
+
+
+/obj/structure/hivemind_machine/node/Destroy()
+	hive_mind_ai.hives.Remove(src)
+	check_for_other()
+	for(var/obj/effect/plant/hivemind/wire in my_wireweeds)
+		remove_wireweed(wire)
+	return ..()
+
+
+/obj/structure/hivemind_machine/node/Process()
+	..()
+	if(!stunned)
+		var/mob/living/carbon/human/target = locate() in view(src)
+		if(target)
+			if(get_dist(src, target) <= 1)
+				icon_state = "core_new-fear"
+			else
+				icon_state = "core_new-see"
+				dir = get_dir(src, target)
+		else
+			icon_state = initial(icon_state)
+
+
+/obj/structure/hivemind_machine/node/update_icon()
+	overlays.Cut()
+	if(stunned)
+		icon_state = "core_new-disabled"
+		overlays += "core-smirk_disabled"
+	else
+		icon_state = initial(icon_state)
+		overlays += "core-smirk"
+
+
+/obj/structure/hivemind_machine/node/use_ability(atom/target)
+	hive_mind_ai.get_points()
+
+
+/obj/structure/hivemind_machine/node/name_pick()
+	name = "[hive_mind_ai.name] [hive_mind_ai.surname]" + " [rand(999)]"
+
+
+//there we binding or un-binding hive with wire
+//in this way, when our node will be destroyed, wireweeds will die too
+/obj/structure/hivemind_machine/node/proc/add_wireweed(obj/effect/plant/hivemind/wireweed)
+	if(wireweed.master_node)
+		wireweed.master_node.remove_wireweed(wireweed)
+	wireweed.master_node = src
+	my_wireweeds.Add(wireweed)
+
+/obj/structure/hivemind_machine/node/proc/remove_wireweed(obj/effect/plant/hivemind/wireweed)
+	my_wireweeds.Remove(wireweed)
+	wireweed.master_node = null
+
+//there we check for other nodes
+//if no any other hives will be found, game over
+/obj/structure/hivemind_machine/node/proc/check_for_other()
+	if(hive_mind_ai)
+		if(!hive_mind_ai.hives.len)
+			hive_mind_ai.die()
+
+
+//TURRET
+//shooting the target with toxic goo
+/obj/structure/hivemind_machine/turret
```

Comment:
I think this turret should be a subtype of /obj/machinery/porta_turret
it has all the fancy logic already built in and will work far better, as well as being more maintainable and less buggy